### PR TITLE
Add Client getUserSettings() request & cache from READY event if applicable

### PR DESCRIPTION
**Pull request by tigercat2000**

### CODE REVIEW

**Comment by Brayzure** [comment]
A bit of an aside, but don't we usually do variables `likeThis` and not `like_this`? Idk if that impacts its ability to be merged.

**Comment by tigercat2000** [comment]
You're right, actually, I wasn't paying attention to the styling.

File: lib/Client.js [hunk 1]
```diff
@@ -1330,6 +1333,15 @@ class Client extends EventEmitter {
     }
 
     /**
+    * Get user settings (user accounts only)
+    * @arg {String} [userID="@me"] The user ID of the target user to get data for. "@me" refers to the logged in user's own settings.
```
Comment (abalabahaha) [inline lib/Client.js]: Pretty sure you can only get your own settings, so this is kinda useless

File: lib/Client.js [hunk 2]
```diff
@@ -146,6 +146,9 @@ class Client extends EventEmitter {
         } else {
             this.voiceConnections = new VoiceConnectionManager();
         }
+
```
Comment (abalabahaha) [inline lib/Client.js]: Can you move these to L141-142

File: lib/Client.js [hunk 3]
```diff
@@ -146,6 +146,9 @@ class Client extends EventEmitter {
...
+        this.userGuildSettings = [];
+        this.userSettings = [];
```
Comment (abalabahaha) [inline lib/Client.js]: If these are objects, it would be better to initialize them as {}

**Review by abalabahaha — CHANGES_REQUESTED** [review]

File: lib/Client.js (hunk 2 above)
Comment (tigercat2000) [reply lib/Client.js]: Sure.

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**